src/components/TheCanvas.vue

```diff
@@ -7,139 +7,140 @@
     @mouseup="onMouseUpHandler"
     @mousemove="throttleToolRender"
   >
-    <v-layer ref="layer" />
+    <v-layer ref="layer"/>
   </v-stage>
 </template>
 
 <script lang="ts">
-import Vue from 'vue'
+/* eslint-disable @typescript-eslint/no-this-alias */
+
 import { Prop } from 'vue-property-decorator'
-import Component, { mixins } from 'vue-class-component'
-import { ITool } from '../types/canvas'
-import Konva from 'konva'
+import Component from 'vue-class-component'
+import { Tool } from '@/tools/Tool'
 import throttle from 'lodash.throttle'
-import { KonvaNodeEvent } from 'konva/types/types'
-const { canvasTools } = require('../mixins/canvasTools')
-
-interface IVmThis {
-    [key: string]: any;
-}
+import Vue from 'vue'
+import { Namespaces } from '@/store'
+import { Action, Getter, namespace } from 'vuex-class'
+import { ToolGetters, ToolsAction } from '@/store/modules/tools'
+import Konva from 'konva'
+import { SocketActions, SocketGetters } from '@/store/modules/socket'
 
-@Component({
-  name: 'TheCanvas',
-  methods: {
-    throttleToolRender: throttle(function (this: any, e:any, websocket: boolean = false) {
-      if (!websocket) {
-        this.onMouseMoveHandler(e)
-      } else {
-        this.onWebSocketHandler(e)
-      }
-    }, 75)
-  }
-})
+const Tools = namespace(Namespaces.TOOLS)
+const Sockets = namespace(Namespaces.SOCKET)
 
-export default class TheCanvas extends mixins(canvasTools) {
-  @Prop() private id!: string;
+  @Component({
+    name: 'TheCanvas'
+  })
+export default class TheCanvas extends Vue {
+    @Prop() private id!: string
+    @Tools.Action(ToolsAction.DISABLE) disable!: () => void
+    @Action(`tools/${ToolsAction.ENABLE}`) enable!: () => void
+    @Action(`tools/${ToolsAction.ENABLE_TOOL}`) enableTool !: (name: string) => void
+    @Getter(`tools/${ToolGetters.ENABLED}`) enabled!: boolean
+    @Getter(`tools/${ToolGetters.TOOLS}`) tools!: Tool[]
+    @Getter(`tools/${ToolGetters.ACTIVE_TOOL}`) activeTool!: Tool
+    @Sockets.Getter(SocketGetters.SOCKET) socket!: WebSocket
+    @Sockets.Action(SocketActions.SEND_IF_OPEN) send!: (message: string) => void
 
-  stageSize = {
-    width: window.innerWidth,
-    height: window.innerHeight
-  }
-  enabledTool: string = ''
-  enabled: boolean = false
-  tools: ITool[] = []
-  $refs!: {
-    layer: any
-  }
+    stageSize = {
+      width: window.innerWidth,
+      height: window.innerHeight
+    }
+    $refs!: {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
isn't @ts-ignore sufficient here?

Reply:
I prefer using eslint disable as ts-ignore does completely bypass the ts compiler